src/charm.py

```diff
@@ -578,13 +580,21 @@ def _alerting_config(self) -> str:
 
         return ",".join(alertmanagers)
 
+    @property
     def _get_v12_migration_date(self) -> str:
-        """Get the 'from' date from the v12 schema in Loki config."""
+        """Get the 'from' date from the v12 schema in Loki config.
+
+        In 2024-05 we changed the storage backend the charm configures from v11/boltdb to v12/tsdb.
+
+        Returns:
+        str: If v12 schema is found, returns the 'from' date of the v12 schema
+                        in ISO 8601 format (YYYY-MM-DD), otherwise returns empty string.
```

Comment:
```suggestion
        Returns:
            The 'from' date of the v12 schema, in YYYY-MM-DD format (ISO 8601), if it is found; otherwise empty string.
```